### PR TITLE
 Change help page document path to fix deployment error   

### PR DESCRIPTION
**Pull request by mimosa2**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Changes made to the path to the help pages to fix an issue with deployment
**Current**
constellation\<module name>\src\au\gov\asd\tac\constellation\<type>\<function>\docs  
**New**
constellation\<module name>\release\modules\ext\docs\<module>\src\au\gov\asd\tac\constellation\<type>\<function>

After the build the documents for each modules are available in the following location
constellation\build\cluster\modules\ext\docs\<module>\src\au\gov\asd\tac\constellation\<type>\<function>

This path should be accessible for the deployed version.  
 
### Why Should This Be In Core?

Help pages for Core repository

### Verification Process

Open Constellation and test access the help pages for any module.

### CODE REVIEW

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java [hunk 1]
```diff
@@ -61,7 +61,8 @@ public Generator() {
     @Override
     public void run() {
         baseDirectory = getBaseDirectory();
-        tocDirectory = String.format("constellation%1$s%2$s", File.separator, TOC_FILE_NAME);
+//        tocDirectory = String.format("constellation%1$s%2$s", File.separator, TOC_FILE_NAME);
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java:64]: remove this commented out line

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java [hunk 2]
```diff
@@ -83,7 +84,8 @@ public void run() {
      * @return a String path for the file location
      */
     public static String getTOCDirectory() {
-        tocDirectory = String.format("constellation%1$s%2$s", File.separator, TOC_FILE_NAME);
+//        tocDirectory = String.format("constellation%1$s%2$s", File.separator, TOC_FILE_NAME);
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java:87]: same here

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java [hunk 3]
```diff
@@ -99,7 +101,9 @@ protected static List<File> getXMLFiles(final String baseDirectory) {
         Lookup.getDefault().lookupAll(HelpPageProvider.class).forEach(provider -> {
             final String providerTOC = provider.getHelpTOC();
             if (StringUtils.isNotEmpty(providerTOC)) {
-                tocXMLFiles.add(new File(baseDirectory + providerTOC));
+//                if (providerTOC.contains("notes-toc.xml") || providerTOC.contains("scripting-toc.xml")) {
+                    tocXMLFiles.add(new File(baseDirectory + providerTOC));
+//                }
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java:106]: same here

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java [hunk 4]
```diff
@@ -117,7 +121,8 @@ public static String getBaseDirectory() {
             // Get the current directory and make the file within the base project directory.
             final String userDir = getResource();
             String[] splitUserDir = userDir.split(Pattern.quote(sep));
-            while (!"constellation".equals(splitUserDir[splitUserDir.length - 1])) {
+//            while (!"constellation".equals(splitUserDir[splitUserDir.length - 1])) {
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/Generator.java:124]: same here

File: CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/GeneratorNGTest.java [hunk 5]
```diff
@@ -80,8 +80,10 @@ public void testRun() {
 
             try (MockedStatic<Generator> generatorStaticMock = Mockito.mockStatic(Generator.class, Mockito.CALLS_REAL_METHODS)) {
                 final List<File> tocXMLFiles = new ArrayList<>();
-                final String layersTOC = "../constellation/src/au/gov/asd/tac/constellation/views/layers/docs/layers-view-toc.xml";
-                final String notesTOC = "../constellation/src/au/gov/asd/tac/constellation/views/notes/docs/notes-view-toc.xml";
+//                final String layersTOC = "../constellation/src/au/gov/asd/tac/constellation/views/layers/docs/layers-view-toc.xml";
+//                final String notesTOC = "../constellation/src/au/gov/asd/tac/constellation/views/notes/docs/notes-view-toc.xml";
```
Comment (antares1470) [inline CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/GeneratorNGTest.java:84]: same here

File: CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/GeneratorNGTest.java [hunk 6]
```diff
@@ -167,7 +169,8 @@ public void testGetBaseDirectory() throws URISyntaxException, MalformedURLExcept
             for (final File file : File.listRoots()) {
 
                 // try within base level of help module
-                final String userDir = file.getPath() + "Users" + sep + "Username" + sep + "Constellation" + sep + "constellation" + sep + "CoreHelp";
+//                final String userDir = file.getPath() + "Users" + sep + "Username" + sep + "Constellation" + sep + "constellation" + sep + "CoreHelp";
```
Comment (antares1470) [inline CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/GeneratorNGTest.java:172]: same here

File: CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/HelpMapperNGTest.java [hunk 7]
```diff
@@ -61,8 +61,10 @@ public void testGetHelpAddress() {
         testGetMappings();
         String className = "au.gov.asd.tac.constellation.help.preferences.HelpOptionsPanelController";
         final String sep = File.separator;
-        String expResult = ".." + sep + "constellation" + sep + "CoreHelp" + sep + "src" + sep + "au" + sep + "gov" + sep
-                + "asd" + sep + "tac" + sep + "constellation" + sep + "help" + sep + "docs" + sep + "help-options.md";
+//        String expResult = ".." + sep + "constellation" + sep + "CoreHelp" + sep + "src" + sep + "au" + sep + "gov" + sep
+//                + "asd" + sep + "tac" + sep + "constellation" + sep + "help" + sep + "docs" + sep + "help-options.md";
```
Comment (antares1470) [inline CoreHelp/test/unit/src/au/gov/asd/tac/constellation/help/utilities/HelpMapperNGTest.java:65]: same here

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]
Looks like it is mostly working. Just a few things that need to be fixed:

- The bootstrap folder with all the CSS styling for the web pages needs to be moved into the ext folder
- The link from Help -> Help Pages in the menu at the top of Constellation has not been updated

**Review by antares1470 — CHANGES_REQUESTED** [review]
From testing, I found a few issues in some of the help pages
- The Graph Window -> hyperlink in page is broken
- The Constellation Graph Model -> image link is broken
- Merge Nodes (the Visual Graph one, not DAV) -> hyperlink in page is broken
- All the Analytics View Questions -> all the hyperlinks in these pages were broken
- Scripting View Links -> it appears the javadoc pages (from hyperlinks) aren't compatible with system (possibly)

File: CoreHelp/src/au/gov/asd/tac/constellation/help/ConstellationHelpDisplayer.java [hunk 8]
```diff
@@ -237,8 +235,8 @@ public boolean display(final HelpCtx helpCtx) {
         final String helpId = helpCtx.getHelpID();
         LOGGER.log(Level.INFO, "display help for: {0}", helpId);
 
-        final String helpDefaultPath = sep + "constellation" + sep + "CoreFunctionality" + sep + "src" + sep + "au" + sep + "gov"
-                + sep + "asd" + sep + "tac" + sep + "constellation" + sep + "functionality" + sep + "docs" + sep + "about-constellation.md";
+        final String helpDefaultPath = "ext" + sep + "docs" + sep + "CoreFunctionality" + sep + "src" + sep + "au" + sep + "gov"
```
Comment (Delphinus8821) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/ConstellationHelpDisplayer.java:238]: This needs to start with a sep before the "ext" for online help

Comment (mimosa2) [reply CoreHelp/src/au/gov/asd/tac/constellation/help/ConstellationHelpDisplayer.java:238]: Ok thanks. We are only fixing for Off line help at the moment.

**Review by antares1470 — APPROVED** [review]

**Review by Delphinus8821 — APPROVED** [review]

**Pull request merged**